neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -275,6 +311,12 @@ private void OnHeadersMessageReceived(HeadersPayload payload)
 
         private void OnInventoryReceived(IInventory inventory)
         {
+            if (inventory.InventoryType == InventoryType.StateRoot && !knownHashes.Contains(inventory.Hash))
```

Comment:
if it is in known hashes, it should not continue and is sent to the task manager.